### ISSUE TITLE
Cannot submit add-on: IntegrityError ... foreign key constraint fails 

### ISSUE THREAD

**Issue by kumar303** [mozilla/olympia#1124]
Caught this from someone who was submitting an add-on. It probably would have shown them a forever spinning progress bar.

https://sentry.prod.mozaws.net/operations/olympia-prod/group/159981/
````
IntegrityError: (1452, 'Cannot add or update a child row: a foreign key constraint fails (`addons_mozilla_org`.`applications_versions`, CONSTRAINT `applications_versions_ibfk_4` FOREIGN KEY (`version_id`) REFERENCES `versions` (`id`) ON DELETE CASCADE)')
Stacktrace (most recent call last):

  File "celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "amo/decorators.py", line 168, in wrapper
    return f(*args, **kw)
  File "amo/decorators.py", line 155, in __call__
    return self.f(*args, **kw)
  File "devhub/tasks.py", line 72, in submit_file
    create_version_for_upload(addon, file_)
  File "amo/decorators.py", line 168, in wrapper
    return f(*args, **kw)
  File "amo/decorators.py", line 155, in __call__
    return self.f(*args, **kw)
  File "amo/decorators.py", line 258, in inner
    return fn(*args, **kwargs)
  File "devhub/tasks.py", line 91, in create_version_for_upload
    version = Version.from_upload(file_, addon, [amo.PLATFORM_ALL.id])
  File "versions/models.py", line 151, in from_upload
    application=app.id).save()
  File "django/db/models/base.py", line 545, in save
    force_update=force_update, update_fields=update_fields)
  File "django/db/models/base.py", line 573, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "django/db/models/base.py", line 654, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "django/db/models/base.py", line 687, in _do_insert
    using=using, raw=raw)
  File "django/db/models/manager.py", line 232, in _insert
    return insert_query(self.model, objs, fields, **kwargs)
  File "django/db/models/query.py", line 1514, in insert_query
    return query.get_compiler(using=using).execute_sql(return_id)
  File "django/db/models/sql/compiler.py", line 903, in execute_sql
    cursor.execute(sql, params)
  File "django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "django/db/backends/mysql/base.py", line 124, in execute
    return self.cursor.execute(query, args)
  File "MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
````

**Comment by kumar303** [comment]
This is the same exception but in a different part of the code https://sentry.prod.mozaws.net/operations/olympia-prod/group/149540/

**Comment by asamuzaK** [comment]
For reference
1225930 – Uploading a new version leads to empty version with no files, adding xpi to version returns a 400 error
https://bugzilla.mozilla.org/show_bug.cgi?id=1225930

**Comment by kumar303** [comment]
Yeah, there is a lot of fallout when exceptions occur because this code wasn't running inside a transaction. Corrupted add-ons should be addressed when https://github.com/mozilla/olympia/pull/1141 is deployed to production. We'll probably need to wrap the version creation views in transactions too but we need to monitor performance first.

**Comment by andymckay** [comment]
Tracker #1244 and this will be fixed with #1141, hopefully.

**Comment by andymckay** [comment]
I believe this is now fixed with the addition of transactions.
